### PR TITLE
[Mailer] Possibility of attaching files to emails

### LINKED ISSUE #3225: [Mailer] Allow to easily add attachments

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | yes |
| Related tickets | closes #3225 |
| License         | MIT |

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -157,6 +157,9 @@ After:
 
 ### Mailer / MailerBundle
 
+* Modified `SenderInterface` and `AdapterInterface` `::send` method, to take an `attachments` array as the last, optional argument.
```
Comment (pamil) [inline UPGRADE.md]:
```
`SenderInterface` and `AdapterInterface` `::send`
```

->

```
`SenderInterface::send` and `AdapterInterface::send`
```

wdyt?

File: UPGRADE.md [hunk 2]
```diff
@@ -157,6 +157,9 @@ After:
...
+* Upgraded `SwiftMailerAdapter` to take the new parameter into account. The `attachments` array should contain absolute paths to the files, you'd like to attach to the emails sent.
```
Comment (pamil) [inline UPGRADE.md]:
```
The `attachments` array should contain absolute paths to the files, you'd like to attach to the emails sent.
```

->

```
The `attachments` array should contain absolute paths to the files being sent with an email.
```

wdyt?

File: src/Sylius/Bundle/MailerBundle/Sender/Adapter/SwiftMailerAdapter.php [hunk 3]
```diff
@@ -57,6 +58,12 @@ public function send(
 
         $message->setBody($renderedEmail->getBody(), 'text/html');
 
+        foreach($attachments as $attachment) {
```
Comment (Zales0123) [inline src/Sylius/Bundle/MailerBundle/Sender/Adapter/SwiftMailerAdapter.php]: Missing space after `foreach` 🌠

File: src/Sylius/Bundle/MailerBundle/Sender/Adapter/SwiftMailerAdapter.php [hunk 4]
```diff
@@ -57,6 +58,12 @@ public function send(
...
+            $file = \Swift_Attachment::fromPath($attachment);
+
+            $message->attach($file);
```
Comment (Zales0123) [inline src/Sylius/Bundle/MailerBundle/Sender/Adapter/SwiftMailerAdapter.php]: Why not just `$massage->attach(\Swift_Attachment::fromPath($attachment));`?

**Comment by pjedrzejewski** [comment]
Thanks Jasiek! :)

File: src/Sylius/Bundle/MailerBundle/Sender/Adapter/SwiftMailerAdapter.php (hunk 4 above)
Comment (roydondsouza) [reply src/Sylius/Bundle/MailerBundle/Sender/Adapter/SwiftMailerAdapter.php:64]: @Zales0123 , @pjedrzejewski , @pamil 
I had an issue, where I was using S3 pre-signed URLs and my file name was showing up with all the extra amazon signature etc from query parameters from the signed URL and Gmail broke my pdf preview as it couldn't recognize the file. 
How can I overwrite the Swift attachment and set the filename, content type, mime type? I know we can use this?
```\Swift_Attachment::fromPath("<S3 PATH>")->setDisposition('inline')->setFilename("<FILENAME>")->setContentType("application/pdf")```


If I send the \SwiftAttachment objects in attachment array by customizing it, this code will override it, how can I overcome that?
```$file = \Swift_Attachment::fromPath($attachment);```

**Pull request merged**